onfleet/onfleet.py

```diff
@@ -168,14 +171,27 @@ def __call__(self, *args, **kwargs):
             json_response = response.json()
 
             if 'code' in json_response:
-                message = json_response['message']
-                cause = message.get('cause', '')
-
-                if isinstance(cause, dict) and cause['type'] == 'duplicateKey':
-                    raise OnfleetDuplicateKeyException("{}: {} (value: '{}', key: '{}')" \
-                        .format(message['error'], message['message'], cause['value'], cause['key']))
-                raise OnfleetException("{}: {} ({})" \
-                    .format(message['error'], message['message'], cause))
+                # So presumably this is an error response,
+                # Map these to better python variable names, since onfleet uses
+                # horrible naming conventions
+                error_type = json_response['code']
+                error_data = json_response['message']
+                error_cause = error_data['cause']
+                error_code = error_data['error']
+                error_message = error_data['message']
+
+                error_string = '{code}: {message} (cause)'.format(
+                    code=error_code,
+                    message=error_message,
+                    cause=error_cause,
+                )
+
+                if error_type == 'duplicateKey':
+                    raise OnfleetDuplicateKeyException(error_string)
+                elif error_cause == 'Geocoding errors found.':
```

Comment:
I may not have investigated this as much as possible. Onfleet is pretty light on the docs. Here is the gist they have of error codes. https://gist.github.com/euskode/8a34b2152cb35ddeb659

If we look at our past errors, we've gotten geocoding errors that have code 1000. I'm led to believe that error 1000 (InvalidContent) could occur under a large number of circumstances.